### PR TITLE
M/concurrent copy from issues

### PR DESCRIPTION
**Pull request by mfelsche**

### CODE REVIEW

File: sql/src/main/java/org/elasticsearch/action/bulk/BulkShardProcessor.java [hunk 1]
```diff
@@ -204,6 +218,45 @@ private ShardId shardId(String indexName, String id, @Nullable String routing) {
...
+                        shardIdFuture.set(localShardId);
+                        clusterService.remove(this);
+                    }
+                }
+            });
+        } else {
+            shardIdFuture.set(shardId);
+        }
+        try {
+            return shardIdFuture.get();
```
Comment (seut) [inline sql/src/main/java/org/elasticsearch/action/bulk/BulkShardProcessor.java]: this must timeout afaik

File: sql/src/main/java/org/elasticsearch/action/bulk/BulkShardProcessor.java [hunk 2]
```diff
@@ -399,8 +454,13 @@ private void processResponse(Response response) {
 
         for (int i = 0; i < response.itemIndices().size(); i++) {
             int location = response.itemIndices().get(i);
+            ShardUpsertResponse.Response innerResp = response.responses().get(i);
+            if (innerResp == null) {
+                ShardUpsertResponse.Failure failure = response.failures().get(i);
+                LOGGER.debug("ShardUpsert Item {} failed: {}", location, failure);
```
Comment (seut) [inline sql/src/main/java/org/elasticsearch/action/bulk/BulkShardProcessor.java]: isn't this an error?

File: stresstest/src/test/java/io/crate/stress/ConcurrentCopyFromTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                "    bytes long,\n" +
+                "    up long,\n" +
+                "    down long,\n" +
+                "    \"end\" timestamp\n" +
+                ") CLUSTERED BY (user_id) INTO 30 SHARDS PARTITIONED BY (begin)\n" +
+                "  WITH (column_policy = 'strict', number_of_replicas=0)");
+        ensureGreen();
+    }
+
+    @Repeat(iterations=10)
```
Comment (seut) [inline stresstest/src/test/java/io/crate/stress/ConcurrentCopyFromTest.java]: is this by intend?

Comment (mfelsche) [inline stresstest/src/test/java/io/crate/stress/ConcurrentCopyFromTest.java]: yep, it's a stresstest :)

Comment (seut) [inline stresstest/src/test/java/io/crate/stress/ConcurrentCopyFromTest.java]: ah yeah missed that :)

**Comment by seut** [comment]
ack

**Pull request merged**